libtenzir/builtins/formats/json.cpp

```diff
@@ -482,14 +482,18 @@ class doc_parser {
 
 auto get_schema_name(simdjson::ondemand::document_reference doc,
                      const selector& selector) -> caf::expected<std::string> {
-  auto type = doc[selector.selector_field];
+  auto object = doc.get_value();
+  for (auto& field : selector.path) {
+    object = object[field];
+  }
   doc.rewind();
-  if (auto err = type.error()) {
-    if (err != simdjson::error_code::NO_SUCH_FIELD)
+  if (auto err = object.error()) {
+    if (err != simdjson::error_code::NO_SUCH_FIELD) {
       return caf::make_error(ec::parse_error, error_message(err));
+    }
     return std::string{unknown_entry_name};
   }
-  auto maybe_schema_name = type.value_unsafe().get_string();
+  auto maybe_schema_name = object.value_unsafe().get_string();
```

Comment:
Haven't tried it yet, but I think this will crash if it's not a string, as that isn't checked above.

Reply:
I think this is fine, because `get_string()` should return an error if it is not a string, which is checked right below.

Hence having for example an integer gives us a `warning: INCORRECT_TYPE: The JSON element does not have the requested type.`. But we could additionally try to convert at least integers to strings here.

Reply:
For windows event logs we probably needs that as the selector field to use is likely the event id.

Reply:
I added an integer -> string conversion as an additional feature to this PR.